RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -0,0 +1,113 @@
+#include "RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace edm;
+
+OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p) {
+  auto cc = setWhatProduced(this);
+
+  transientBS_ = new BeamSpotObjects;
+  theHLTBS_ = new BeamSpotOnlineObjects;
+  theLegacyBS_ = new BeamSpotOnlineObjects;
+  fakeBS_ = new BeamSpotOnlineObjects;
+  fakeBS_->SetBeamWidthX(0.1);
+  fakeBS_->SetBeamWidthY(0.1);
+  fakeBS_->SetSigmaZ(15.);
+  fakeBS_->SetPosition(0., 0., 0.);
+  fakeBS_->SetType(-1);
+
+  bsHLTToken_ = cc.consumesFrom<BeamSpotOnlineObjects, BeamSpotOnlineHLTObjectsRcd>();
+  bsLegacyToken_ = cc.consumesFrom<BeamSpotOnlineObjects, BeamSpotOnlineLegacyObjectsRcd>();
+}
+
+void OnlineBeamSpotESProducer::fillDescription(edm::ParameterSetDescription& desc) {}
+
+const BeamSpotOnlineObjects* OnlineBeamSpotESProducer::compareBS(const BeamSpotOnlineObjects* bs1,
+                                                                 const BeamSpotOnlineObjects* bs2) {
+  //Random logic so far ...
+  if (bs1->GetSigmaZ() - 0.0001 < bs2->GetSigmaZ()) {  //just temporary for debugging
+    if (bs1->GetSigmaZ() > 5.) {
+      return bs1;
+    } else {
+      return bs2;
+    }
+
+  } else {
+    if (bs2->GetSigmaZ() > 5.) {
+      return bs2;
+    } else {
+      return bs1;
+    }
+  }
+}
+
+OnlineBeamSpotESProducer::~OnlineBeamSpotESProducer() {
+  delete theHLTBS_;
+  delete theLegacyBS_;
+  delete transientBS_;
```

Comment:
double delete here as well, transientBS_ is set to theHLTBS_ or theLegacyBS_  so it deletes one of those twice.